device/store.go

```diff
@@ -49,6 +50,10 @@ func (d *Device) getDbDir() string {
 }
 
 func (d *Device) migrateDb() error {
+	if d.db == nil {
```

Comment:
Is this truly necessary? The db is initialized when creating a new device, so it should not be `nil`. This applies also to all other changes in this file. If that is a first step towards making SQLite optional, maybe it might be better to move it in a new PR.

Reply:
It is the latter, but in general I found it more consistent to put it here, so if there are no strong objections I'd keep it in this PR